### PR TITLE
Refactored Feature to include Geometry type as type param.

### PR DESCRIPTION
**Pull request by lossyrob**
This PR changes `abstract class Feature[D] { type G <: Geometry }` into `case class Feature[G <: Geometry, D](...)`. This fell out of me wanting to write a simple `mapData` method, and the current type architecture not allowing me to. This mostly affects internal code; the API should remain the same as long as users use things like `PointFeature[D]`, etc.

### CODE REVIEW

**Comment by lossyrob** [comment]
@echeipesh please review.

File: vector/src/main/scala/geotrellis/vector/reproject/package.scala [hunk 1]
```diff
@@ -3,6 +3,8 @@ package geotrellis.vector
 import geotrellis.proj4._
 
 package object reproject {
+  type DI = DummyImplicit
```
Comment (echeipesh) [inline vector/src/main/scala/geotrellis/vector/reproject/package.scala]: Looks like this was needed when trying an overload but not with renames of apply?

**Comment by echeipesh** [comment]
+1, shame I didn't think to implement maps to see this.

File: vector/src/main/scala/geotrellis/vector/reproject/package.scala (hunk 1 above)
Comment (lossyrob) [inline vector/src/main/scala/geotrellis/vector/reproject/package.scala]: Good call!

**Pull request merged**